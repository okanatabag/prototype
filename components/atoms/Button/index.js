import styled from "styled-components";

const Button = styled.button`
	font-size: ${(props) => (props.size ? `${props.size}rem` : "2rem")};
	padding: 4px 8px;
	cursor: pointer;
	color: ${(props) => props.tc || "#000"};
	background: ${(props) => (props.bg ? props.bg : "#fff")};
	border-radius: ${(props) => (props.rounded ? "50%" : "5px")};
	@media only screen and (max-width: 520px) {
		font-size: ${(props) =>
			props.size ? `${props.size * 0.8}rem` : "1.6rem"};
	}
	@media only screen and (max-width: 460px) {
		font-size: ${(props) =>
			props.size ? `${props.size * 0.6}rem` : "1.2rem"};
	}
	@media only screen and (max-width: 360px) {
		font-size: ${(props) =>
			props.size ? `${props.size * 0.5}rem` : "1rem"};
	}
`;

export default Button;
