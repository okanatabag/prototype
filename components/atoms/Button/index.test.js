import React from "react";
import renderer from "react-test-renderer";
import Button from ".";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <Button size="2" tc="#ffffff" bg="#000000">
        Test
      </Button>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
