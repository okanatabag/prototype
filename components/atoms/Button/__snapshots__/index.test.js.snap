// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="Button-sc-dqgx36-0 idTkMv"
  size="2"
>
  Test
</button>
`;
